### PR TITLE
Anst/fixes and improvements

### PR DESCRIPTION
**Pull request by achimmihca**
### What does this PR do?

**Small changes**
- MicSampleRecorder: fix NPE if MicProfile not set
- SongVideoPlayer: better error handling
    - a notification with the error message is shown (similar to an error when playing audio). Furthermore video playback is not attempted again for this SongMeta.
- fix highscore not beeing saved when scene is finished automatically
    - when Unity finishes an audio clip, the position in the song is reset to 0, which is why it was not recognized as "you have sung all notes in the song".
- fix video issue when song is longer than the video
    - when Unity finishes a video clip, the position of the video is reset to 0. This caused flickering. Now, the video just pauses when the song is at a position where there is no video anymore.
- added SongRouletteItemContextMenuHandler to open song folder in system explorer (right click on a song in the SongSelectScene to see the context menu).
    - very useful if you have to correct encoding or fix audio / video of a song
- removed VersionLabel from LoadingScene and ApplicationUtils.AppVersion. They are obsolete because of the new VERSION.txt file
- set default cursor in project's player settings back to none. Otherwise it is not possible to use the system's default cursor

- SongSelectScene: perform fuzzy search on song title *without white-space*. Now you can type a song title without space-bar, which is more intuitive I think.
- SongSelectScene: scroll-wheel now scrolls in the same direction as in USDX
- SongSelectScene: hold left/right arrow key for continued scrolling
    - The ContinuedKeyPressDetector and all KeyboardInputControllers will probably be replaced with Unity's new input system when it will be released. So I would not bother with this too much.

**Bigger changes**
- Log and skip unkown file extensions (#166)
    - I tried some different formats. Only mp3, ogg and wav worked for audio. Unity was not able to 
play flac, aac, aiff and so on. Also some of my avi videos work and some don't. What seems to work best are ogg and mp4 files.

- Save the player score when switching between SingScene and SongEditor. Scoring is continued where the scene was left. So you are still not able to gain more than 10000 points (except you modify the song significantly, but that would be cheating).
    - Therefore, the player score data has been separated as an own class, namely PlayerScoreControllerData. 
    - To avoid confusion, I renamed PlayerScoreData to PlayerScoreResultData, which already existed for the SingingResultsScene
    - This feature makes it possible to go to the SongEditor to fix small issues and then return to the SingScene (possibly 1 or 2 sentences before it was left) to continue playing. This way, small issues in songs can be ironed out while playing (e.g. wrong lyrics or some notes that are a bit too long). Such a feature was a wish that I had for USDX.

### Closes Issue(s)

#166 

### Motivation

While playing a bit with USPlay, I found some usability issues and errors that I want to address with this PR.

### CODE REVIEW

File: UltraStar Play/Assets/Scenes/Sing/SingSceneController.cs [hunk 1]
```diff
@@ -108,7 +108,15 @@ void Start()
             {
                 playerProfilesWithoutMic.Add(playerProfile);
             }
-            CreatePlayerController(playerProfile, micProfile);
+            PlayerController playerController = CreatePlayerController(playerProfile, micProfile);
+
+            if (SceneData.PlayerProfileToScoreDataMap.Count > 0)
+            {
+                if (SceneData.PlayerProfileToScoreDataMap.TryGetValue(playerProfile, out PlayerScoreControllerData scoreData))
```
Comment (basisbit) [inline UltraStar Play/Assets/Scenes/Sing/SingSceneController.cs:115]: any reason for those two if's not being one if?

File: UltraStar Play/Assets/Common/Scene/SceneData/PlayerScoreControllerData.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        Sentence = sentence;
+    }
+}
+
+[Serializable]
+public class NoteScore
+{
+    public Note Note { get; private set; }
+    public int correctlySungBeats { get; set; }
```
Comment (basisbit) [inline UltraStar Play/Assets/Common/Scene/SceneData/PlayerScoreControllerData.cs:34]: shouldn't this be PascalCase?

Comment (achimmihca) [reply UltraStar Play/Assets/Common/Scene/SceneData/PlayerScoreControllerData.cs]: Yes. Copy&paste mistake.

File: UltraStar Play/Assets/Scenes/Sing/SingSceneController.cs (hunk 1 above)
Comment (achimmihca) [reply UltraStar Play/Assets/Scenes/Sing/SingSceneController.cs]: Nope. The first if is just useless. I guess at some point, there was more in this if-branch.

File: UltraStar Play/Assets/Common/Video/SongVideoPlayer/SongVideoPlayer.cs [hunk 3]
```diff
@@ -260,14 +278,25 @@ private void InitVideo(SongMeta songMeta)
...
     void OnDisable()
     {
+        videoPlayer.errorReceived -= OnVideoPlayerErrorReceived;
         if (HasLoadedVideo)
         {
             ClearOutRenderTexture(videoPlayer.targetTexture);
         }
     }
 
+    private void OnVideoPlayerErrorReceived(VideoPlayer source, string message)
```
Comment (basisbit) [inline UltraStar Play/Assets/Common/Video/SongVideoPlayer/SongVideoPlayer.cs:295]: Codacy Issue found: [Remove this unused method parameter 'source'.](https://app.codacy.com/manual/UltraStar-Deluxe/Play/pullRequest?prid=5520139)

File: UltraStar Play/Assets/Scenes/SongSelect/SongRoulette/SongRouletteItemContextMenuHandler.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+using UnityEngine.UI;
+using UniInject;
+using UniRx;
+using System.IO;
+
+public class SongRouletteItemContextMenuHandler : AbstractContextMenuHandler
+{
+    public SongMeta SongMeta { get; set; }
+
+    protected override void FillContextMenu(ContextMenu contextMenu)
```
Comment (basisbit) [inline UltraStar Play/Assets/Scenes/SongSelect/SongRoulette/SongRouletteItemContextMenuHandler.cs:15]: Codacy Issue found: [Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation.](https://app.codacy.com/manual/UltraStar-Deluxe/Play/pullRequest?prid=5520139)

File: UltraStar Play/Assets/Common/Input/ContinuedKeyPressDetector.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+using UniRx;
+
+public class ContinuedKeyPressDetector : MonoBehaviour
+{
+    public KeyCode keyCode;
+
+    public float initialDelayInSeconds = 0.75f;
+    public float repeatedDelayInSeconds = 0.25f;
+
+    private Subject<ContinuedKeyPressEvent> continuedKeyPressEventStream = new Subject<ContinuedKeyPressEvent>();
```
Comment (basisbit) [inline UltraStar Play/Assets/Common/Input/ContinuedKeyPressDetector.cs:17]: Codacy Issue found: [Make 'continuedKeyPressEventStream' 'readonly'.](https://app.codacy.com/manual/UltraStar-Deluxe/Play/pullRequest?prid=5520139)

File: UltraStar Play/Assets/Scenes/Sing/PlayerController/PlayerScoreController.cs [hunk 6]
```diff
@@ -135,39 +135,44 @@ private void OnBeatAnalyzed(PlayerPitchTracker.BeatAnalyzedEvent beatAnalyzedEve
...
+            return;
+        }
+
         Note analyzedNote = noteAnalyzedEvent.Note;
-        if (noteToNoteScoreMap.TryGetValue(analyzedNote, out NoteScore noteScore))
+        if (ScoreData.NoteToNoteScoreMap.TryGetValue(analyzedNote, out NoteScore noteScore))
         {
             //Debug.Log($"OnNoteAnalyzed: {noteScore.correctlySungBeats} / {analyzedNote.Length}, {analyzedNote.StartBeat}, {analyzedNote.EndBeat}, {analyzedNote.Text}");
-            if (noteScore.correctlySungBeats >= analyzedNote.Length)
+            if (noteScore.CorrectlySungBeats >= analyzedNote.Length)
```
Comment (basisbit) [inline UltraStar Play/Assets/Scenes/Sing/PlayerController/PlayerScoreController.cs:175]: Codacy Issue found: [Merge this if statement with the enclosing one.](https://app.codacy.com/manual/UltraStar-Deluxe/Play/pullRequest?prid=5520139)

**Comment by basisbit** [comment]
Ah great, codacy now is extra annoying with the recent changes...

**Pull request merged**